dokka-integration-tests/gradle/build.gradle.kts

```diff
@@ -1,43 +1,208 @@
  */
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
+import org.gradle.api.tasks.testing.logging.TestLogEvent.FAILED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.SKIPPED
+import org.gradle.language.base.plugins.LifecycleBasePlugin.VERIFICATION_GROUP
+import org.jetbrains.kotlin.gradle.dsl.ExplicitApiMode.Disabled
 
 plugins {
-    id("dokkabuild.test-integration")
+    id("dokkabuild.kotlin-jvm")
+    `test-suite-base`
+    `java-test-fixtures`
 }
 
 dependencies {
-    implementation(projects.utilities)
+    api(projects.utilities)
 
-    implementation(kotlin("test-junit5"))
-    implementation(libs.junit.jupiterApi)
-    implementation(libs.junit.jupiterParams)
+    api(libs.jsoup)
 
-    implementation(gradleTestKit())
+    api(libs.kotlin.test)
+    api(libs.junit.jupiterApi)
+    api(libs.junit.jupiterParams)
 
-    implementation(libs.jsoup)
+    api(gradleTestKit())
 }
 
 kotlin {
     // this project only contains test utils and isn't published, so it doesn't matter about explicit API
-    explicitApi = ExplicitApiMode.Disabled
+    explicitApi = Disabled
+
+    compilerOptions {
+        allWarningsAsErrors = false
+        optIn.add("kotlin.io.path.ExperimentalPathApi")
+    }
 }
 
 val aggregatingProject = gradle.includedBuild("dokka")
+val templateProjectsDir = layout.projectDirectory.dir("projects")
 
-tasks.integrationTest {
+tasks.integrationTestPreparation {
+    // TODO remove this in https://github.com/Kotlin/dokka/pull/3433
     dependsOn(aggregatingProject.task(":publishToMavenLocal"))
+}
+
+tasks.withType<Test>().configureEach {
+    dependsOn(tasks.integrationTestPreparation)
+
+    setForkEvery(1)
+    maxHeapSize = "2G"
+    dokkaBuild.integrationTestParallelism.orNull?.let { parallelism ->
+        maxParallelForks = parallelism
+    }
+
+    val useK2 = dokkaBuild.integrationTestUseK2.get()
+
+    useJUnitPlatform {
+        if (useK2) excludeTags("onlyDescriptors", "onlyDescriptorsMPP")
+    }
+
+    systemProperty("org.jetbrains.dokka.experimental.tryK2", useK2)
+    // allow inspecting projects in temporary dirs after a test fails
+    systemProperty(
+        "junit.jupiter.tempdir.cleanup.mode.default",
+        dokkaBuild.isCI.map { isCi -> if (isCi) "ALWAYS" else "ON_SUCCESS" }.get(),
+    )
 
     environment("DOKKA_VERSION", project.version)
+    environment("isExhaustive", dokkaBuild.integrationTestExhaustive)
+    environment("ANDROID_HOME", dokkaBuild.androidSdkDir.get().invariantSeparatorsPath)
+
+    testLogging {
+        exceptionFormat = FULL
+        events(SKIPPED, FAILED)
+        showExceptions = true
+        showCauses = true
+        showStackTraces = true
+    }
+
+    // TODO remove this in https://github.com/Kotlin/dokka/pull/3433
+    doNotTrackState("uses artifacts from Maven Local")
+}
+
+@Suppress("UnstableApiUsage")
+testing {
+    suites {
+        withType<JvmTestSuite>().configureEach {
```

Comment:
minor: configuration here can be moved inside `registerTestProjectSuite` to easier understand all configuration. Not sure, that we will have any other suite which is not created by it

Reply:
I'd prefer to keep JVM-specific config in the `withType<JvmTestSuite>().configureEach { }` block and test-project specific config in the `registerTestProjectSuite()` function. That's a clear split to me, but maybe that's not clear in the code and I can add a comment to explain?

Could you take another look, now that I've moved `registerTestProjectSuite()` to be a top-level function? I think it should help with readability.

Reply:
now it looks better :)
Could you still add a comment there, regarding why we need this configuration (specifically `implementation(project)`), with link to Gradle docs or something (it's not super obvious behaviour, as I've written above)